tajo-core/src/main/java/org/apache/tajo/engine/planner/PhysicalPlannerImpl.java

```diff
@@ -854,6 +855,14 @@ private PhysicalExec createSortBasedColumnPartitionStorePlan(TaskAttemptContext
           sortSpecs[i++] = new SortSpec(insertNode.getProjectedSchema().getColumn(id), true, false);
         }
       }
+    } else if (storeTableNode.getType() == NodeType.CREATE_TABLE) {
+      int i = 0;
+      for (int j = 0; j < partitionKeyColumns.length; j++) {
+        int id = storeTableNode.getOutSchema().getColumns().size() + j;
```

Comment:
In 'CREATE TABLE', it is to hard to use qualified names of partition keys.

CreateTableNode's in/out schema only contains physically stored columns and does not contain logical columns (i.e., partition key columns). The way described in the patch will work correctly. 

But, since this way is not intuitive, we need to refactor CreateTableNode and InsertTableNode to provide a better way to access physical columns corresponding to partition keys.